File: services/language-service/src/main/java/dev/animedia/languageservice/application/service/DeleteLanguageService.java
package dev.animedia.languageservice.application.service;

import dev.animedia.languageservice.application.usecase.DeleteLanguageUseCase;
import dev.animedia.languageservice.domain.exception.LanguageNotFoundException;
import dev.animedia.languageservice.domain.repository.LanguageCommandRepository;
import dev.animedia.languageservice.domain.repository.LanguageQueryRepository;

public class DeleteLanguageService implements DeleteLanguageUseCase {
    private final LanguageCommandRepository languageCommandRepository;
    private final LanguageQueryRepository languageQueryRepository;

    public DeleteLanguageService(
        LanguageCommandRepository languageCommandRepository,
        LanguageQueryRepository languageQueryRepository
    ) {
        this.languageCommandRepository = languageCommandRepository;
        this.languageQueryRepository = languageQueryRepository;
    }

    @Override
    public void delete(String code) {
        languageQueryRepository.findByCode(code)
            .orElseThrow(LanguageNotFoundException::new);
        languageCommandRepository.delete(code);
    }
}

File: services/social-service/src/core/Application/Language/Service/Private/DeleteLanguageService.php
<?php

declare(strict_types=1);

namespace Core\Application\Language\Service\Private;

use Core\Application\Language\Exception\LanguageNotFoundException;
use Core\Application\Language\UseCase\Private\DeleteLanguageUseCase;
use Core\Domain\Language\Repository\LanguageCommandRepositoryInterface;
use Core\Domain\Language\Repository\LanguageQueryRepositoryInterface;

final readonly class DeleteLanguageService implements DeleteLanguageUseCase
{
    public function __construct(
        private LanguageQueryRepositoryInterface $languageQueryRepository,
        private LanguageCommandRepositoryInterface $languageCommandRepository
    ) {}

    #[\Override]
    public function execute(string $languageIsoCode): void
    {
        $exists = $this->languageQueryRepository->existsByIsoCode($languageIsoCode);
        if (!$exists) {
            throw new LanguageNotFoundException($languageIsoCode);
        }

        $this->languageCommandRepository->delete($languageIsoCode);
    }
}
